src/util/amount.rs

```diff
@@ -1208,6 +1213,15 @@ mod tests {
         assert_eq!(sp("-200000000000 BTC"), Err(E::TooBig));
         assert_eq!(p("18446744073709551616 sat"), Err(E::TooBig));
 
+        assert_eq!(sp("0 msat"), Err(E::TooPrecise));
+        assert_eq!(sp("-0 msat"), Err(E::TooPrecise));
+        assert_eq!(sp("000 msat"), Err(E::TooPrecise));
+        assert_eq!(sp("-000 msat"), Err(E::TooPrecise));
+        assert_eq!(p("0 msat"), Err(E::TooPrecise));
+        assert_eq!(p("-0 msat"), Err(E::TooPrecise));
```

Comment:
Should this be `TooPrecise` or `Negative`?

Reply:
Yeah I had this dilemma as well. I decided that if two things are wrong, it doesn't matter which one you report; so one should just return the first one that is detected. In this case the negative check is deferred until later, so it's faster to return the `TooPrecise` case.

Reply:
To make the test more correct one could expect **either** a `Negative` or `TooPrecise` error, so that the tests don't break if the implementation would change the check order.

Reply:
Nah, we can fix the test when/if it breaks. It's good to notice changed behavior like this anyway, in case users are depending on it.